### PR TITLE
fix(rpc): return error instead of wrong operation type in otterscan API

### PR DESCRIPTION
**Pull request by emhane**
Return error instead of wrong operation type for EOFCREATE in otterscan API, until otterscan API has EOF support. Likely users will believe that there is smthg wrong with the node rather than RPC response if we return the wrong operation type.

### CODE REVIEW

**Comment by emhane** [comment]
blocked by https://github.com/alloy-rs/alloy/issues/1288

File: crates/rpc/rpc/src/otterscan.rs [hunk 1]
```diff
@@ -120,6 +120,7 @@ where
                             TransferKind::Create => OperationType::OpCreate,
                             TransferKind::Create2 => OperationType::OpCreate2,
                             TransferKind::SelfDestruct => OperationType::OpSelfDestruct,
+                            TransferKind::EofCreate => todo!(),
```
Comment (DaniPopes) [inline crates/rpc/rpc/src/otterscan.rs:123]: map to `OpCreate`?

**Comment by DaniPopes** [comment]
Doesn't have to be blocked on it, I doubt otterscan will add a new variant for this since it's essentially the same as create

File: crates/rpc/rpc/src/otterscan.rs (hunk 1 above)
Comment (emhane) [reply crates/rpc/rpc/src/otterscan.rs:123]: the code here is just formatting an RPC response about what happened on the blockchain, not using the operation in execution. we do need to block or panic.

Comment (emhane) [reply crates/rpc/rpc/src/otterscan.rs:123]: otterscan rpc is used in otterscan block scanner right?

Comment (emhane) [reply crates/rpc/rpc/src/otterscan.rs]: returning unimplemented error for now

**Review by mattsse — COMMENTED** [review]
I don't think this is useful because mostlikely the user doesn't care whether the create was done in eof or not, as far as otterscan is concerned this is also just a Create and we shouldn't fail, so mapping this to a regular create variant makes more sense here until there's a spec update for otterscan

**Comment by emhane** [comment]
ref https://github.com/otterscan/otterscan/issues/2457

**Comment by mattsse** [comment]
cool then we can add the variant to the otterscan type in alloy

**Pull request closed without merging**